src/tools/kata-ctl/src/arch/mod.rs

```diff
@@ -33,6 +33,17 @@ pub use x86_64 as arch_specific;
 )))]
 compile_error!("unknown architecture");
 
-pub fn check() -> Result<()> {
-    arch_specific::check()
+pub fn get_arch_type() -> String {
+
+    let machine_kind = if cfg!(target_arch = "x86_64") {
```

Comment:
nit: I would say use
```Rust
if cond1 {
    "str1"
} else if cond2 {
    "str2"
}.to_string()
```
instead of 
```Rust
let _str = if cond1 {
    "str1"
} else if cond2 {
    "str2"
};
_str.to_string()
```

Reply:
@Ji-Xinyou thank you for your feedback, the` machine_kind` variable was removed because we opt to define a `CheckItem` list per architecture.